src/main/kotlin/dev/minn/jda/ktx/InlineShardManagerBuilder.kt

```diff
@@ -0,0 +1,542 @@
+    
+    var audioSendFactory: IAudioSendFactory? = null
+        set(value) {
+            builder.setAudioSendFactory(value)
+            field = value
+        }
+    
+    var autoReconnect: Boolean = true
+        set(value) {
+            builder.setAutoReconnect(value)
+            field = value
+        }
+    
+    var bulkDeleteSplitting: Boolean = true
+        set(value) {
+            builder.setBulkDeleteSplittingEnabled(value)
+            field = value
+        }
+    
+    var enableShutdownHook: Boolean = true
+        set(value) {
+            builder.setEnableShutdownHook(value)
+            field = value
+        }
+    
+    @Deprecated("Use eventManagerProvider instead")
+    var eventManager: IEventManager? = null
+        set(value) {
+            if (value != null) {
+                eventManagerProvider = IntFunction { value }
+            }
+            field = value
+        }
+    
+    var eventManagerProvider: IntFunction<out IEventManager>? = null
+        set(value) {
+            if (value != null) {
+                builder.setEventManagerProvider(value)
+            }
+            field = value
+        }
+    
+    var activity: Activity? = null
+        set(value) {
+            builder.setActivity(value)
+            field = value
+        }
+    
+    var activityProvider: IntFunction<Activity>? = null
+        set(value) {
+            builder.setActivityProvider(value)
+            field = value
+        }
+    
+    var idle: Boolean = false
+        set(value) {
+            builder.setIdle(value)
+            field = value
+        }
+    
+    var idleProvider: IntFunction<Boolean>? = null
+        set(value) {
+            builder.setIdleProvider(value)
+            field = value
+        }
+    
+    var status: OnlineStatus? = null
+        set(value) {
+            if (value != null) {
+                builder.setStatus(value)
+            }
+            field = value
+        }
+    
+    var statusProvider: IntFunction<OnlineStatus>? = null
+        set(value) {
+            if (value != null) {
+                builder.setStatusProvider(value)
+            }
+            field = value
+        }
+    
+    var threadFactory: ThreadFactory? = null
+        set(value) {
+            builder.setThreadFactory(value)
+            field = value
+        }
+    
+    var httpClientBuilder: OkHttpClient.Builder? = null
+        set(value) {
+            builder.setHttpClientBuilder(value)
+            field = value
+        }
+    
+    var httpClient: OkHttpClient? = null
+        set(value) {
+            builder.setHttpClient(value)
+            field = value
+        }
+    
+    var rateLimitPool: ScheduledExecutorService? = null
+        set(value) {
+            builder.setRateLimitPool(value, true)
+            field = value
+        }
+    
+    var rateLimitPoolProvider: ThreadPoolProvider<out ScheduledExecutorService>? = null
```

Comment:
Better:

```kotlin
rateLimitPool(false, scheduler)

rateLimitPool(false) {
  newPool()
}
```

Reply:
While making this I forgot about all about kotlin's fancy lambda functions. Will add those changes to everything.